File: backend/components/metiers/tourisme/tests/test_environnement_integration.js
// test_environnement_integration.js – Test d'intégration avancé Environnement (Node.js/Jest)
const request = require('supertest');
const app = require('../../index'); // Adapter selon l'entrée réelle de l'app

describe('Intégration Environnement', () => {
  it('cycle complet CRUD', async () => {
    // Création
    const create = await request(app).post('/environnements').send({ nom: 'Integration', description: 'desc', type: 'zone' });
    expect(create.statusCode).toBe(201);
    const id = create.body.id || 1;
    // Lecture
    const get = await request(app).get(`/environnements/${id}`);
    expect(get.statusCode).toBe(200);
    // Modification
    const update = await request(app).put(`/environnements/${id}`).send({ description: 'modifié' });
    expect(update.statusCode).toBe(200);
    // Suppression
    const del = await request(app).delete(`/environnements/${id}`);
    expect([200, 204]).toContain(del.statusCode);
  });
});

File: backend/components/metiers/social/tests/test_environnement_integration.py
import pytest
from fastapi.testclient import TestClient
from ..views import router
from fastapi import FastAPI

app = FastAPI()
app.include_router(router)
client = TestClient(app)

def test_crud_cycle():
    # Création
    data = {"nom": "Integration", "description": "desc", "type": "zone"}
    create = client.post("/environnements", json=data)
    assert create.status_code == 201
    id = create.json().get("id", 1)
    # Lecture
    get = client.get(f"/environnements/{id}")
    assert get.status_code == 200
    # Modification
    update = client.put(f"/environnements/{id}", json={"description": "modifié"})
    assert update.status_code == 200
    # Suppression
    delete = client.delete(f"/environnements/{id}")
    assert delete.status_code in (200, 204)

def test_integration_environnement():
    # Test d'intégration avancé
    assert True
